sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -152,16 +152,29 @@ def fit(self, X, y):
 
 class NoSparseClassifier(BaseBadClassifier):
     def fit(self, X, y):
-        X, y = check_X_y(X, y, accept_sparse=['csr', 'csc'])
+        X, y = check_X_y(X, y, accept_sparse=['csr', 'csc', 'dok', 'lil',
```

Comment:
I extended `NoSparseClassifier` to accept all kind of sparse (we want to test the error messages in all kind of sparsity no?), and I made it return an Exception instead (wrong type of error), and I created another one (`NoSparseClassifierTypeError`) that returns a TypeError (good type of error) but with the wrong message.

Then I replaced the pre-existing test in `test_check_estimator` that just tested the first part of the message (`Estimator %s doesn't seem to fail gracefully on sparse data`), by two tests that use the two above toy objects and test the whole message in the two cases. 
Is that correct ?